### PR TITLE
Add solution(quiz): problem 101 Equal IV

### PR DESCRIPTION
**Pull request by abhijeet-saxena**

### CODE REVIEW

File: quiz/equal-iv_en.md [hunk 1]
```diff
@@ -1,4 +1,31 @@
...
+## Explanation
+
+The [equality (==) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality) checks whether its two operands are equal, and unlike the strict equality operator ===, it attempts to convert and compare operands that are of different types. We have,
+
+```
+[0].toString() // "0"
+0..toString() // "0"
+```
+
+In this example, we are comparing different data types which means `==` will convert the operands to strings, and then compare. So, the first comparison evaluates to `false`, and the second one is `true`
```
Comment (BFEdev) [inline quiz/equal-iv_en.md]: > In this example, we are comparing different data types which means `==` will convert the operands to strings, and then compare

This doesn't seem right according to ECMAScript spec, the code block below is not exactly the logic under the hood

**Review by BFEdev — CHANGES_REQUESTED** [review]
Thanks for the contribution, but this solution doesn't seem good enough to get merged

File: quiz/equal-iv_en.md (hunk 1 above)
Comment (abhijeet-saxena) [reply quiz/equal-iv_en.md]: Yup. I think I misunderstood the comparison and oversimplified it a lot. I've updated it with more details.

File: quiz/equal-iv_en.md [hunk 2]
```diff
@@ -1,4 +1,34 @@
...
+```
+false
+true
+```
+
+## Explanation
+
+Here, in both cases the left operand `[0]` is an `object` while the right operand is a [primitive](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) (string in first line and number in the second)
+
+The [equality (==) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality) checks whether its two operands are equal, and unlike the strict equality operator ===, it attempts to convert and compare operands that are of different types.  To do this comparison, JS will follow the rules for [Loose equality using ==](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#loose_equality_using) which states that:
```
Comment (BFEdev) [inline quiz/equal-iv_en.md:21]: MDN is second-hand spec source, would you mind switch the MDN links to ECMA spec links ? like this https://tc39.es/ecma262/multipage/abstract-operations.html#sec-islooselyequal

File: quiz/equal-iv_en.md [hunk 3]
```diff
@@ -1,4 +1,34 @@
...
+
+> If one of the operands is an object and the other is a primitive, convert the object to a primitive.
+
+This [Primitive coercion](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#primitive_coercion) effectively uses `toString` to get the primitive value i.e `[0]` will be treated as `"0"`
```
Comment (BFEdev) [inline quiz/equal-iv_en.md:25]: This is over simplified for how [primitive values are generated](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-toprimitive), please add more details.


> `[0]` will be treated as `"0"`

And this is not a simple process, please take a look at our solution for another question [here](https://bigfrontend.dev/quiz/Implicit-Conversion-II/solution) (last section)

**Review by BFEdev — CHANGES_REQUESTED** [review]

**Comment by abhijeet-saxena** [comment]
I think this will require much deeper research and resource collection. Closing this PR in the meantime

**Pull request closed without merging**